src/aspnetcore/Ark.Tools.AspNetCore/JsonContext/ArkProblemDetailsJsonSerializerContext.cs

```diff
@@ -0,0 +1,22 @@
+// Copyright (C) 2024 Ark Energy S.r.l. All rights reserved.
+// Licensed under the MIT License. See LICENSE file for license information. 
+using Ark.Tools.AspNetCore.ProblemDetails;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Ark.Tools.AspNetCore.JsonContext;
+
+/// <summary>
+/// JSON serialization source generation context for Ark ProblemDetails types.
+/// This context includes all common error response types used by Ark.Tools middleware.
+/// Configured with Ark default settings via constructor.
+/// </summary>
+[JsonSerializable(typeof(FluentValidationProblemDetails), TypeInfoPropertyName = "FluentValidationProblemDetails")]
+[JsonSerializable(typeof(FluentValidationErrors), TypeInfoPropertyName = "FluentValidationErrors")]
+[JsonSerializable(typeof(Microsoft.AspNetCore.Mvc.ProblemDetails), TypeInfoPropertyName = "ProblemDetails")]
+[JsonSerializable(typeof(Dictionary<string, FluentValidationErrors[]>), TypeInfoPropertyName = "DictionaryStringFluentValidationErrors")]
+[JsonSerializable(typeof(Dictionary<string, object>), TypeInfoPropertyName = "DictionaryStringObject")]
+public sealed partial class ArkProblemDetailsJsonSerializerContext : JsonSerializerContext
```

Comment:
Add JsonSerializationOptions attribute mimiking the Ark default serialization options in Ark.Tools.SystemTextJson

Do the same for the ProblemDetailsContext in the library ark.tools.aspnetcore

If supported, derive an attribute from JsonSerializationOptions and use it. Check on ms docs MCP.

Add to Agents.md to use MS docs mcp for c# patterns.